app/components/RangeSlider.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useState } from 'react';
+import tinygradient from 'tinygradient';
+
+interface RangeSliderProps {
+    min: number;
+    max: number;
+    step: number;
+    initialValue?: number;
+    onChange: (value: number) => void;
+}
+
+const RangeSlider: React.FC<RangeSliderProps> = ({ min, max, step, initialValue = min, onChange }) => {
+    const [value, setValue] = useState<number>(initialValue);
+
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const newValue = Number(event.target.value);
+        setValue(newValue);
+        onChange(newValue);
+    };
+
+    const percentage = ((value - min) / (max - min)) * 100;
+
+    // gradient for progress track
+    const gradient = tinygradient("#90E568", "#FFDC6A", "#FD6376");
+    const currentColor = gradient.rgbAt(percentage / 100).toHexString();
+
+    const sliderStyle = {
+        background: `linear-gradient(to right, ${gradient.rgbAt(0).toHexString()} 0%, ${currentColor} ${percentage}%, white ${percentage}%, white 100%)`,
+        height: '15px',
+        border: '1px solid',
+        borderRadius: '16px',
+        appearance: 'none',
+        cursor: 'pointer',
+        position: 'relative',
+    };
+
+    const thumbStyle = {
+        width: '15px',
+        height: '45px',
+        backgroundColor: currentColor, // so the thumb changes color with the progress track
+        borderRadius: '16px',
+        border: 'none',
+        cursor: 'pointer',
+        position: 'relative',
+    };
+
+    return (
+        <div className="bg-gray-950 p-10 rounded-lg shadow-lg">
+            <div className="slider-container relative">
+                <input
+                    type="range"
+                    min={min}
+                    max={max}
+                    step={step}
+                    value={value}
+                    onChange={handleChange}
+                    className="w-full"
+                    style={sliderStyle}
+                    style={{ ...sliderStyle, '--thumb-background': currentColor }}
```

Comment:
sliderStyle looks like its being added twice, also consider dynamically generating the styles in one place instead of in several areas (e.g. the thumb background) through the code